src/document.rs

```diff
@@ -314,14 +328,18 @@ impl Document {
                 .unwrap_or(default_excerpt_separator);
 
             let excerpt = if let Some(excerpt_str) = excerpt_attr {
-                excerpt_str
+                excerpt_str.to_string()
             } else if excerpt_separator.is_empty() {
-                ""
+                String::new()
```

Comment:
Curses! :)

Your question about the errors if I use the `try!(...)` in each clause led me to the solution.  Basically: my issue was I was leaving the semicolon terminating the `if let Some(excerpt_str) = ...` expression causing a `mismatched types` error.

Once I fixed that everything seems to work!  I'll push the new changes for review.